validator/lib/validators.js

```diff
@@ -0,0 +1,114 @@
+import { CID } from 'multiformats/cid'
+import { sha256 } from 'multiformats/hashes/sha2'
+import { CarBlockIterator } from '@ipld/car'
+import { LinkIndexer } from 'linkdex'
+import { maybeDecode } from 'linkdex/decode.js'
+import { equals } from 'uint8arrays'
+import * as raw from 'multiformats/codecs/raw'
+import * as pb from '@ipld/dag-pb'
+
+/**
+ * Parse and validate the CID
+ *
+ * @param {string} cid
+ * @returns {{errors: *[], info: CID<any, number, number, Version>}}
+ */
+export function parseCid (cid) {
+  let errors = []
+  let info
+  try {
+    info = CID.parse(cid)
+  } catch (err) {
+    errors = [{ cid, detail: 'Unable to parse cid', err }]
+  }
+
+  return { info, errors }
+}
+
+export const MAX_BLOCK_SIZE = 1 << 21 // 2MiB
+
+export class InvalidCarError extends Error {
+  /**
+   * @param {string} reason
+   */
+  constructor (reason) {
+    super(`Invalid CAR file received: ${reason}`)
+    this.name = 'InvalidCar'
+    this.status = 400
+    this.code = InvalidCarError.CODE
+  }
+}
+InvalidCarError.CODE = 'ERROR_INVALID_CAR'
+
+export async function carStats (carBytes) {
```

Comment:
we don't need to do the all the same validation that the api does here... kubo has done most of this for us already. We really only need to detect CAR truncation here.